src/Symfony/Component/AssetMapper/Tests/Factory/MappedAssetFactoryTest.php

```diff
@@ -121,6 +122,32 @@ public function compile(string $content, MappedAsset $asset, AssetMapperInterfac
         $this->assertSame('7e4f24ebddd4ab2a3bcf0d89270b9f30', $asset->digest);
     }
 
+    /**
+     * @dataProvider provideCreateMappedAssetHashAlgorithms
+     */
+    public function testCreateMappedAssetHashAlgorithms(?string $hashAlgorithm, string $expectedDigest)
+    {
+        $factory = $this->createFactory(null, $hashAlgorithm);
+        $asset = $factory->createMappedAsset('subdir/file6.js', __DIR__.'/../Fixtures/dir2/subdir/file6.js');
+        $this->assertSame($expectedDigest, $asset->digest);
+    }
+
+    public static function provideCreateMappedAssetHashAlgorithms()
+    {
+        yield 'none' => [null, '7f983f4053a57f07551fed6099c0da4e'];
```

Comment:
looking at the changes made for the `createFactory()` method, I wonder if moving this into a dedicated test where we don't use `createFactory()` wouldn't increase the readability of the tests

Reply:
👍  i think it's easier to read now

Reply:
(I mean, after I rewrote the tests following your comment :) )